### DISCUSSION TITLE
Resolving realm by using input form

### DISCUSSION THREAD

**Discussion by MarcelSz** [Finbuckle/Finbuckle.MultiTenant#595] [Q&A]
Hello.

I have a Blazor WebAssembly app and I'm trying to integrate it with KeyCloak, which has multiple tenants setup. I'd like to have a specific strategy such that the user's realm is determined by his email provided in some UI form. Realm mapping (username or email -> realm id) would be retrieved from the database, but this is an implementation detail. Saying shortly, it should be a realm resolution based on email address. 

In Azure OIDC there is a "/common" authority,which exactly matches my case and everything is done on the Microsoft's OIDC side, but apparently I can use KeyCloak only.

As of now, I produced a solution, which gets the realm from the subdomain on my own (without using this library), but unfortunately putting customer's name is "a no go" for the team. I also tried to produce a "auth broker" Blazor solution, but I'm failing, because of IAccessTokenProvider and AccessTokenNotAvailableException have hidden fields in the 'original' page.

Yesterday I had found this library and it looks very promising. I know that I have a very specific case, but still I'm looking for any tips. Maybe this library would help me with my case?

Thanks in advance for any response.

**Comment by AndrewTriesToCode** [comment]
Hi, I think the library might be helpful for you. If you look at the .NET Core 3.1 samples there is one for "Shared Signin" where the initial input is user email and from that I determine the user tenant, sign them in, and redirect them. That sounds like it might be close to what you are looking for. If you have any more specific questions after looking into it let me know.

**Reply by MarcelSz** [reply]
Thanks for the response. I cloned the code and I was looking for this particular example, but unfortunately couldn't find the "Shared Signin" example. I was checking other samples and found similar one which is _PerTenantAuthenticationSample_. However it is path based too :-(. Besides this, I had to enable Implicit Flow in my KeyCloak OIDC.
